src/expt_file_counts.py

```diff
@@ -0,0 +1,646 @@
+
+    return constructed_filter
+
+def get_experiments_filter(filter_dict, constructed_filter):
+    if not isinstance(filter_dict, dict):
+        msg = f'Invalid type for filter, must be \'dict\', actually ' \
+            f'type: {type(filter_dict)}'
+        raise TypeError(msg)
+    
+    if not isinstance(constructed_filter, dict):
+        msg = 'Invalid type for constructed_filter, must be \'dict\', ' \
+            f'actually type: {type(filter_dict)}'
+        raise TypeError(msg)   
+    
+    constructed_filter = get_string_filter(
+        filter_dict, exp, 'name', constructed_filter, 'experiment_name')
+    
+    constructed_filter = get_time_filter(
+        filter_dict, exp, 'cycle_start', constructed_filter)
+
+    constructed_filter = get_time_filter(
+        filter_dict, exp, 'cycle_stop', constructed_filter)
+
+    constructed_filter = get_time_filter(
+        filter_dict, exp, 'wallclock_start', constructed_filter)
+    
+    return constructed_filter
+
+def get_file_types_filter(filter_dict, constructed_filter):
+    constructed_filter = get_string_filter(
+        filter_dict, ft, 'name', constructed_filter, 'file_type_name')
+
+    constructed_filter = get_string_filter(
+        filter_dict, ft, 'file_template', constructed_filter, 'file_template')
+
+    constructed_filter = get_string_filter(
+        filter_dict, ft, 'file_format', constructed_filter, 'file_format')
+    
+    return constructed_filter
+
+def get_storage_locations_filter(filter_dict, constructed_filter):
+    constructed_filter = get_string_filter(
+        filter_dict, sl, 'name', constructed_filter, 'storage_location_name')
+
+    constructed_filter = get_string_filter(
+        filter_dict, sl, 'bucket_name', constructed_filter, 'bucket_name')
+
+    constructed_filter = get_string_filter(
+        filter_dict, sl, 'platform', constructed_filter, 'platform')
+
+    constructed_filter = get_string_filter(
+        filter_dict, sl, 'platform_region', constructed_filter, 'platform_region')
+    
+    constructed_filter = get_string_filter(
+        filter_dict, sl, 'key', constructed_filter, 'key')
+    
+    return constructed_filter
+
+
+def get_experiment_id(experiment_name, wallclock_start):
+    expt_request = {
+        'name': 'experiment',
+        'method': db_utils.HTTP_GET,
+        'params': {
+            'filters': {
+                'name': {
+                    'exact': experiment_name
+                },
+                'wallclock_start': {
+                    'exact': wallclock_start
+                },
+            },
+            'ordering': [
+                {'name': 'wallclock_start', 'order_by': 'desc'}
+            ],
+            'record_limit': 1
+        }
+    }
+    print(f'expt_request: {expt_request}')
+    er = ExperimentRequest(expt_request)
+    results = er.submit()
+    print(f'results: {results}')
+
+    record_cnt = 0
+    try:
```

Comment:
as discussed in person, this would be a larger rewrite and most of the code block could not be separated into smaller catches for this logic. something to be considered for a future clean up of the repo.